moped-toolbox/council_districts_load/insert_council_districts.py

```diff
@@ -0,0 +1,124 @@
+
+MUTATION = """
+mutation InsertCouncilDistricts($objects: [layer_council_district_insert_input!]!) {
+  insert_layer_council_district(objects: $objects) {
+    returning {
+      id
+    }
+  }
+}
+"""
+
+
+def make_hasura_request(*, query, env, variables=None):
+    """Fetch data from hasura
+
+    Args:
+        query (str): the hasura query
+        variables (dict): optional variables to be included
+        env (str): the environment name, which will be used to acces secrets
+
+    Raises:
+        ValueError: If no data is returned
+
+    Returns:
+        dict: Hasura JSON response data
+    """
+    admin_secret = HASURA["hasura_graphql_admin_secret"][env]
+    endpoint = HASURA["hasura_graphql_endpoint"][env]
+    headers = {
+        "X-Hasura-Admin-Secret": admin_secret,
+        "content-type": "application/json",
+    }
+    payload = {"query": query, "variables": variables}
+    res = requests.post(endpoint, json=payload, headers=headers)
+    res.raise_for_status()
+    data = res.json()
+    try:
+        return data["data"]
+    except KeyError:
+        raise ValueError(data)
+
+
+def get_council_districts_geojson():
+    res = requests.get(DISTRICTS_URL)
+    res.raise_for_status()
+    return res.json()["features"]
+
+
+def make_multipoly(features):
+    for feature in features:
+        if feature["geometry"]["type"] == "Polygon":
+            feature["geometry"]["type"] = "MultiPolygon"
+            feature["geometry"]["coordinates"] = [feature["geometry"]["coordinates"]]
+            print(
+                f"Converted district {feature['properties']['COUNCIL_DISTRICT']} to MuliPolygon"
+            )
+
+def merge_district_2(features):
+    district_2s = [f for f in features if f["properties"]["COUNCIL_DISTRICT"] == 2]
+    other_districts = [f for f in features if f["properties"]["COUNCIL_DISTRICT"] != 2]
+    new_district_2_coordinates = []
+    # combine geometries of all district 2's
+    for f in district_2s:
+        if f["geometry"]["type"] != "MultiPolygon":
+            raise Exception("Feature must be multipolygon for this hack to work")
+        new_district_2_coordinates = new_district_2_coordinates + f["geometry"]["coordinates"]
+    # update the first district 2 with the geometry of all district 2s
+    new_district_2 = district_2s[0]
+    new_district_2["geometry"]["coordinates"] = new_district_2_coordinates
+    # add this back into our feature array
+    other_districts.append(new_district_2)
+    return other_districts
```

Comment:
as complained about [here](https://austininnovation.slack.com/archives/C015VKQSA23/p1674750864318159), the layer has two district 2 polygons 😑

Reply:
@frankhereford i'd love to hear your thoughts on this.

i also wonder if we should accept the AGOL features as-is and just have two district 2s? part of me thinks it's best to keep the data as close to the authoritative layer as possible. it really won't affect any of our querying. my biggest concern is that we'd have to drop the `unique` constraint from the `council_district` column and that certainly invites future trouble.

Reply:
My heart wants to go with whatever clean-up needing to be done be done in this script, and then this script can become an infrequently run ETL to keep the data up to date. I can totally see a world where this becomes a spin-off issue to avoid blocking getting something out the door that is pressing.

Reply:
Also, I can see that districts 1, 2, & 5 are interesting in that they have wholly disjointed, disconnected areas (think an "island off the coast" of the main district area, which may explain why they are malformed, but certainly does not excuse it.